stdlib/public/core/MutableCollection.swift

```diff
@@ -364,6 +364,50 @@ extension MutableCollection where SubSequence == Slice<Self> {
   }
 }
 
+//===----------------------------------------------------------------------===//
+// moveSubranges(_:to:)
+//===----------------------------------------------------------------------===//
+
+#if !$Embedded
+extension MutableCollection {
+  /// Moves the elements in the given subranges to just before the element at
+  /// the specified index.
+  ///
+  /// This example finds all the uppercase letters in the array and then
+  /// moves them to between `"i"` and `"j"`.
+  ///
+  ///     var letters = Array("ABCdeFGhijkLMNOp")
+  ///     let uppercaseRanges = letters.subranges(where: { $0.isUppercase })
+  ///     let rangeOfUppercase = letters.moveSubranges(uppercaseRanges, to: 10)
+  ///     // String(letters) == "dehiABCFGLMNOjkp"
+  ///     // rangeOfUppercase == 4..<13
+  ///
+  /// - Parameters:
+  ///   - subranges: The subranges of the elements to move.
+  ///   - insertionPoint: The index to use as the destination of the elements.
+  /// - Returns: The new bounds of the moved elements.
+  ///
+  /// - Complexity: O(*n* log *n*) where *n* is the length of the collection.
+  @available(SwiftStdlib 5.11, *)
+  @discardableResult
+  public mutating func moveSubranges(
```

Comment:
This can't currently be `@inlinable` because `_indexedStablePartition` is `internal` and not `@usableFromInline`, so making this `@inlinable` would require exposing an ABI entry point for `_indexedStablePartition`

Reply:
That's unfortunate; it aptly demonstrates the pain of not having ABI entry points. 😞 One solution would be to mark `_indexedStablePartition` as `@backDeployed(before: 5.11) internal`. But I suppose we aren't pressed to do this yet -- we can leave this as potential future work.